### DISCUSSION TITLE
The JS class keyword

### DISCUSSION THREAD

**Discussion by JonEBee** [nextgenhealthcare/connect#4703] [Feature/Enhancement Ideas]
Mirth 3.8.x on the Linux platform
JavaScript now allows the creation of object prototypes using the class keyword.  When will we be able to use that keyword within the script editors?  The editors flag a "reserved keyword" error.

Thanks.

**Comment by pacmano1** [comment]
There are other similar errors. I would create an issue if I were you. They may actually be an already open one.￼  I'd look for that first and comment on it.￼

**Comment by tonygermano** [comment]
Mirth only supports what is supported by the Rhino library. The `class` keyword is not currently implemented. There is an existing issue for it, but I wouldn't expect it any time soon. It likely won't be in Rhino for a while, and mirth is usually not running the latest version, anyway. https://github.com/mozilla/rhino/issues/835.

**Reply by tonygermano** [reply]
ES5 style classes still work just fine, though, as well as [Object.create()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create).
